samples/guestbook/aws/provision_db/main.go

```diff
@@ -0,0 +1,113 @@
+// Copyright 2018 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// The provision_db program connects to an RDS database and initializes it with
+// SQL from stdin. It is intended to be invoked from Terraform.
+package main
+
+import (
+	"fmt"
+	"io"
+	"io/ioutil"
+	"log"
+	"net/http"
+	"os"
+	"os/exec"
+	"path/filepath"
+	"strings"
+)
+
+func main() {
+	if len(os.Args) != 5 {
+		fmt.Fprintf(os.Stderr, "usage: provision_db HOST SECURITY_GROUP DATABASE ROOT_PASSWORD\n")
+		os.Exit(64)
+	}
+	log.SetPrefix("aws/provision_db: ")
+	if err := provisionDb(os.Args[1], os.Args[2], os.Args[3], os.Args[4]); err != nil {
+		log.Fatal(err)
+	}
+}
+
+func provisionDb(dbHost, securityGroupID, dbName, dbPassword string) error {
+	mySQLImage := "mysql:5.6"
+
+	// Pull the necessary Docker images.
+	log.Print("Downloading Docker images...")
+	if _, err := run("docker", "pull", mySQLImage); err != nil {
+		return err
+	}
+
+	// Create a temporary directory to hold the certificates.
+	// We resolve all symlinks to avoid Docker on Mac issues, see
```

Comment:
Except that this is not resolving the symlinks, so it would have the compatibility issue mentioned.

Reply:
Done.